pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -702,35 +703,51 @@ func getInstallerPodImageFromEnv() string {
 }
 
 // ensureRequiredResourcesExist makes sure that all non-optional resources are ready or it will return an error to trigger a requeue so that we try again.
-func (c InstallerController) ensureRequiredResourcesExist() error {
+func (c InstallerController) ensureRequiredResourcesExist(revisionNumber int32) error {
 	missingSecrets := []string{}
 	missingConfigs := []string{}
 
+	configMapsNames := sets.NewString()
+	for _, c := range c.configMaps {
+		configMapsNames.Insert(c.Name)
+	}
 	for _, cm := range append(append([]revision.RevisionResource{}, c.certConfigMaps...), c.configMaps...) {
 		if cm.Optional {
 			continue
 		}
-		_, err := c.configMapsGetter.ConfigMaps(c.targetNamespace).Get(cm.Name, metav1.GetOptions{})
+		name := cm.Name
```

Comment:
I would rather split the for loop. This does not make the logic easier.

Reply:
i don't like duplicating the logic just because the name in one case need suffix.

Reply:
would adding comment here help?

Reply:
We have that loop already two times. Having it 4 times does not make it worse.